src/python/zquantum/core/circuit/gates/_single_qubit_gates.py

```diff
@@ -0,0 +1,156 @@
+"""Gates acting on single qubit."""
+from abc import ABC
+from typing import Union
+import sympy
+import numpy as np
+from ._gate import SpecializedGate
+
+
+class SingleQubitGate(SpecializedGate, ABC):
+    """Base class for single qubit gates.
+
+    Attributes:
+        qubit: index of qubit this gate acts on.
+    Args:
+        qubit: index of qubit this gate acts on.
+    """
+
+    def __init__(self, qubit: int):
+        self.qubit = qubit
+        super().__init__((qubit,))
+
+
+class X(SingleQubitGate):
+    """Quantum X gate."""
+
+    def _create_matrix(self) -> sympy.Matrix:
+        return sympy.Matrix([[0, 1], [1, 0]])
+
+
+class Y(SingleQubitGate):
+    """Quantum Y gate."""
+
+    def _create_matrix(self) -> sympy.Matrix:
+        return sympy.Matrix([[0, -1.0j], [1.0j, 0.0]])
+
+
+class Z(SingleQubitGate):
+    """Quantum Z gate."""
+
+    def _create_matrix(self) -> sympy.Matrix:
+        return sympy.Matrix([[1, 0], [0, -1]])
+
+
+class H(SingleQubitGate):
+    """Quantum Hadamard gate."""
+
+    def _create_matrix(self) -> sympy.Matrix:
+        return sympy.Matrix(
+            [
+                [(1 / np.sqrt(2)), (1 / np.sqrt(2))],
+                [(1 / np.sqrt(2)), (-1 / np.sqrt(2))],
+            ]
+        )
+
+
+class I(SingleQubitGate):
+    """Quantum Identity gate."""
+
+    def _create_matrix(self) -> sympy.Matrix:
+        return sympy.Matrix([[1, 0], [0, 1]])
+
+
+class PHASE(SingleQubitGate):
+    """Quantum Phase gate."""
+
+    def _create_matrix(self) -> sympy.Matrix:
+        return sympy.Matrix(
+            [
+                [1, 0],
+                [0, complex(0, 1)],
```

Comment:
Let's be consistent with the previous (`Y`) and use either `complex` or the `1.0j` notation

Reply:
You're right, fixed in 5987cf8